include/exadg/incompressible_navier_stokes/postprocessor/output_generator.cpp

```diff
@@ -186,6 +242,7 @@ OutputGenerator<dim, Number>::evaluate(
   VectorType const &                                                    velocity,
   VectorType const &                                                    pressure,
   std::vector<dealii::SmartPointer<SolutionField<dim, Number>>> const & additional_fields,
+  std::vector<dealii::SmartPointer<SolutionField<dim, Number>>> const & surface_fields,
```

Comment:
It seems we currently have a mixture of designs. In some sense, this line breaks the modularity of the Navier-Stokes postprocessing design (we need to change interfaces to realize more/other functionality). According to the current function interface above, I would have expected that the new functionality is included in the existing `write_output()` function. Instead, a separate `write_surface_output()` function is introduced, which aims at a modular design somewhat deeper in the code. However, this raises the question why surface output is then not realized completely modular as a separate postprocessing module (e.g. `SurfaceOutputGenerator`, or a seperate function inside the existing `OutputGenerator`). In my opinion, both realizations are possible and should be discussed.

Some arguments below:

Another option could be to equip `SolutionField` with additional information (volume vs. surface) such that the if-branches can be made at the point where output is written. I am not sure if this makes the design better, because in the end, we also have to discuss the design in relation to the deal.II library (which has separate classes `DataOut` and `DataOutFaces`). This argument relates also to this discussion https://github.com/exadg/exadg/pull/349/files#r1126093917

What should be mentioned at this point and be part of the overall discussion is that I discussed with @c-p-schmidt some time ago that it would actually be better that `write_output()` would only receive `additional_fields` as argument and not `velocity` and `pressure` separately, because this would allow us to (maybe) use exactly the same `write_output()` function for all PDE modules in ExaDG (Poisson, ConvDiff, NavierStokes, etc.). This argument supports to keep `write_output()` and `write_surface_output()` separately (and separate arguments `volume_fields` and `surface_fields`), as is done currently.

Reply:
I'm not sure whether it works for all modules but it sounded very appealing to me. If this `additional_fields` data object consists of objects the final output routine is able to write to disc, then the `write_output()` method should mainly need to iterate over the provided data from `additional_fields` and choose the correct output method to write to disc.

Reply:
Not passing in `velocity` and `pressure` (and hence not needing the two  dof_handlers) and reusing the same `write_output` sounds quite elegant. This calls for some `Calculators` in `navier_stokes_calculators` then, right? I think adding the fields and then having surface and volume output separate in `utilities` or `postprocessor` would be good.

Reply:
calculators should not be necessary, because velocity/pressure are the primary variables for IncNS. One just needs to add velocity/pressure to the vector of solution fields.